File: NEsper/events/TypedEventPropertyGetter.cs
using System;
namespace net.esper.events
{
	
	/// <summary> Interface for property getters also returning type information for the property.</summary>
	/// <author>  pablo
	/// </author>
	public interface TypedEventPropertyGetter:EventPropertyGetter
	{
		/// <summary> Returns type of event property.</summary>
		/// <returns> class of the objects returned by this getter
		/// </returns>
		Type ResultClass
		{
			get;
			
		}
	}
}

File: trunk/esper/src/main/java/com/espertech/esper/event/TypedEventPropertyGetter.java
package com.espertech.esper.event;

/**
 * Interface for property getters also returning type information for the property.
 * @author pablo
 */
public interface TypedEventPropertyGetter extends EventPropertyGetter {

	/**
	 * Returns type of event property.
	 * @return class of the objects returned by this getter
	 */
	public Class getResultClass();
}
